src/SupabaseClient.ts

```diff
@@ -131,12 +131,11 @@ export default class SupabaseClient {
   }
 
   /**
-   * Remove all active subscriptions.
+   * Remove all subscriptions.
    */
-  removeAllSubscriptions() {
-    this.realtime.channels.forEach((sub) => {
-      this.removeSubscription(sub)
-    })
+  async removeAllSubscriptions() {
+    const subscriptions = this.realtime.channels.slice()
+    await Promise.allSettled(subscriptions.map((sub) => this.removeSubscription(sub)))
```

Comment:
let's return something from is function so that our users can `await` the connections to all be closed - useful for cleanups

Reply:
@kiwicopple yep, that was my intention! forgot the `return`